course_management.lua

```diff
@@ -150,9 +152,11 @@ function courseplay:loadCourse(vehicle, id, useRealId, addCourseAtEnd) -- fn is
 			local numCourse1, numCourse2 = #course1, #course2;
 			local course1wp, course2wp = numCourse1, 1;
 
-			--find crossing points, merge at first pair where dist < 50
-			local firstMatchFound, closestMatchFound = false, false;
-			local useFirstMatch = false; --true: first match <50m is used to merge / false: match with closest distance <50m is used to merge;
+			local matchFound = false 
+			local wpDistMax = 30 --suggest reduction from 50 
```

Comment:
Rather leave this at 50

Reply:
Yes, your considerations are valid, yet I do not like hard-coded value like 45 degrees. 

This is supposed to work on all intersections, not just straight angle. So if two roads merge at 45 degree angle then one turn will be just 45 angle, but another very valid direction change will be 135 degrees where hard-coded 45 degrees will fail to make a connection at all.

You should also keep in mind that vector angle difference between two desired waypoints is not the whole story. Course direction at the intersection actually changes twice - at first waypoint from stored direction to calculated direction and at second way point from calculated to stored. It may well be that first change is 70 degrees and another just 20 degrees. That is why I actually add these two together to arrive to a decision. 

There is also a very real possibility that best crossing waypoint is actually past second course and you have to turn back (more than 90 degrees) anyway.  

What I can do is to make process a two-step. I could search for two combinations - one that is minimum angle within 30 m, another - minimum distance where angle does not exceed your preference of 45 (or, maybe, 90 total) within 50 m. Only use former when cannot find later.  

How does that sound?